src/zenml/integrations/great_expectations/utils.py

```diff
@@ -45,11 +46,17 @@ def create_batch_request(
     Returns:
         A Great Expectations runtime batch request.
     """
-    # get pipeline name, step name and run id
-    step_env = cast(StepEnvironment, Environment()[STEP_ENVIRONMENT_NAME])
-    pipeline_name = step_env.pipeline_name
-    run_id = step_env.pipeline_run_id
-    step_name = step_env.step_name
+    try:
+        # get pipeline name, step name and run id
+        step_env = cast(StepEnvironment, Environment()[STEP_ENVIRONMENT_NAME])
+        pipeline_name = step_env.pipeline_name
+        run_id = step_env.pipeline_run_id
+        step_name = step_env.step_name
+    except KeyError:
+        # if not running inside a pipeline step, use random values
+        pipeline_name = f"pipeline_{random_str(5)}"
+        run_id = f"pipeline_{random_str(5)}"
+        step_name = f"step_{random_str(5)}"
```

Comment:
Is overlapping ids and names etc a problem? Do we need to check that they don't already exist?

Reply:
These are only used to generate identifiers for temporary resources that don't survive the execution of a single data profiling or data validation call. The chance that a clash will happen is almost zero and the Great Expectations library has builtin checks for it anyway.

Besides, this isn't really meant to be used outside a pipeline run, merely offered as a convenience for testing GE locally outside of a pipeline (i.e. no parallelism involved). But you have a good point to inquire, thanks for bringing it up !